### DISCUSSION TITLE
[Enhancement] [input] 增强文本输入框字符长度验证根据字节验证长度

### DISCUSSION THREAD

**Discussion by liwenbo112** [element-plus/element-plus#11081] [Q&A] [answered]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `input`

## Description

需要字符长度根据字节显示，一个中文对应2个字节一个字母对应一个字节 
示例："我的name" 应为 8个长度

### Additional comments

_(empty)_

**Comment by liwenbo112** [comment]
@rotu

**Comment by chenxch** [comment, marked as answer]
中文长度计算为2个字符是程序员之间涉及数据库存储时候的认知，对于普通用户群里来说并非如此，反而会觉得是有问题。如果你需要可以自己计算，然后通过suffix slots 显示。

**Comment by liwenbo112** [comment]
好的，谢谢


啊哈
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "element-plus/element-plus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年12月27日(星期二) 中午1:34
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [element-plus/element-plus] [Enhancement] [input] 增强文本输入框字符长度验证根据字节验证长度 (Discussion #11081)





 
中文长度计算为2个字符是程序员之间涉及数据库存储时候的认知，对于普通用户群里来说并非如此，反而会觉得是有问题。如果你需要可以自己计算，然后通过suffix slots 显示。
